lib/ecto_crux.ex

```diff
@@ -559,32 +564,57 @@ defmodule EctoCrux do
       defp crux_paginate(%Ecto.Query{} = query, %{page: page} = opts)
            when is_integer(page) and page > 0 do
         page_size = crux_page_size(opts)
-        do_crux_paginate(query, page_size * (page - 1), opts)
+        total_entries = count(query, to_keyword(opts))
+        total_pages = crux_total_pages(total_entries, page_size)
+
+        meta = %{
+          page_size: page_size,
+          total_entries: total_entries,
+          total_pages: total_pages,
+          page: if(page > total_pages, do: total_pages, else: page)
+        }
+
+        do_crux_paginate(query, page_to_offset(meta.page, page_size), meta)
```

Comment:
We would gain in simplicity by only having the logic with the `meta` object written once, by replacing this function's content with :
```
page_size = crux_page_size(opts)
new_opts = opts
|> Map.put(page_to_offset(page, page_size))
|> Map.delete(:page)
crux_paginate(query, new_opts)
```

Reply:
There is two different case, one for `:page `option the other one for` :offset `option. Respectively, we need to calculate, in the first case, the offset from the page and on the other one the page number from the offset. But finally both are calling `do_crux_paginate `function. I created the meta map like to be clear on what will be returned from the function and avoid Map manipulation, and more importantly to avoid computing again the same thing (page_size, total_entries, total_pages) later. As is it, every computation for meta data are made once.

TLDR; I don't understand the content of the function in you suggestion

Reply:
Yes, I understood these cases. However, by doing so, you write twice the logic of setting the max to the `offset`/`page`, and of creating the `meta` object. My proposal was that to transform the `:page ` case so that it simply translates its arguments to the `:offset` case, and then calls it. Then, no more logic duplication, at the cost of a (tiny) bit more computation.